web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+
+        initConfiguration();
+        clearQueue();
+
+        ArrayList<TelemetryItem> expectedTelemetries = sentHttpGetRequests();
+        ArrayList<TelemetryItem> realTelemetries = getTelemetryFromAzure(DocumentType.Requests);
+
+        ArrayList<TelemetryItem> missingTelemetry = getMissingExpectedTelemetry(realTelemetries, expectedTelemetries);
+        if (missingTelemetry.size() > 0) {
+            for (TelemetryItem item : missingTelemetry) {
+                System.out.println("Didn't find matching item in real telemetry for request of URI " + item.getProperty("uri"));
+            }
+
+            throw new Exception("Didn't find match for " + missingTelemetry.size() + " items");
+        }
+        else {
+            System.out.println("Test passed successfully");
+        }
+    }
+
+    private void initConfiguration() throws JAXBException {
+        String location = System.getenv(configEnvironmentVariable);
+        Map<String, String> env =  System.getenv();
+        System.out.println("Configuration file located in " + location);
+
+        try {
+           config = IntegrationTestConfiguration.load(location);
+        }
+        catch(JAXBException e) {
+            System.out.println("Failed to load configuration file " + location);
+            throw e;
+        }
+    }
+
+    /**
+     * Clears the Azure queue from all existing messages
+     * @throws Exception
+     */
+    private void clearQueue() throws Exception {
+        CloudStorageAccount account = CloudStorageAccount.parse(config.getStorageConnectionString());
+        CloudQueueClient queueClient = account.createCloudQueueClient();
+        CloudQueue queue = queueClient.getQueueReference(config.getQueueName());
+
+        queue.downloadAttributes();
+        System.out.println("Before clearing: Queue contains " + queue.getApproximateMessageCount() + " items");
+        System.out.println("Clearing queue ...");
+        queue.clear();
+    }
+
+    private ArrayList<TelemetryItem> sentHttpGetRequests() throws Exception {
+
+        String serverAddress = config.getTestServerAddress();
+        int tomcat7Port = config.getTomcat7Port();
+        String applicationFollder = config.getApplicationFolder();
+
+        String requestId1 = UUID.randomUUID().toString();
+        String requestId2 = UUID.randomUUID().toString();
+        String requestId3 = UUID.randomUUID().toString();
+
+        URI uri1 = new URI("http://" + serverAddress + ":" + tomcat7Port + "/" + applicationFollder + "/books?id=Thriller&runId=" + runId + "&requestId=" + requestId1);
+        URI uri2 = new URI("http://" + serverAddress + ":" + tomcat7Port + "/" + applicationFollder + "/loan?title=Gone%20Girl&id=030758836x&subject=Thriller&runId=" + runId + "&requestId=" + requestId2);
+        URI uri3 = new URI("http://" + serverAddress + ":" + tomcat7Port + "/" + applicationFollder + "/notExistingWebFage?runId=" + runId + "&requestId=" + requestId3);
+
```

Comment:
You're have the same URL construction logic in the above 3 lines.
Why not refactoring it into a method, i.e. 
URI constructUrl(server, port, app, path) {
    return new URI("http://" + server + ":" + port + "/" + app + path);
}